ivan-a-kovtun/lab1-BasicIO/user.c

```diff
@@ -0,0 +1,134 @@
+/******************************************************************************/
+/* Files to Include                                                           */
+/******************************************************************************/
+
+#ifdef __XC32
+#include <xc.h>          /* Defines special function registers, CP0 regs  */
+#endif
+
+#include <stdint.h>          /* For uint32_t definition                       */
+#include <stdbool.h>         /* For true/false definition                     */
+#include "user.h"            /* variables/params used by user.c               */
+
+/******************************************************************************/
+/* User Functions                                                             */
+
+/******************************************************************************/
+
+void InitApp(void) {
+    /* Setup analog functionality and port direction */
+
+    // LED output
+    // Disable analog mode for G6
+    ANSELGbits.ANSG6 = 0;
+    // Set direction to output for G6
+    TRISGbits.TRISG6 = 0;
+
+    // Initialize outputs for other LEDs
+    ANSELBbits.ANSB11 = 0;
+    TRISBbits.TRISB11 = 0;
+    
+    ANSELGbits.ANSG15 = 0;
+    TRISGbits.TRISG15 = 0;
+    
+    /* ANSELDbits.ANSD4 = 0;
+       we don't write this string because
+       it is no analog mode for port D
+     */
+    TRISDbits.TRISD4 = 0;
+
+    // Turn on LEDs for testing
+    LD1_PORT_BIT = 1;
+    LD2_PORT_BIT = 1;
+    LD3_PORT_BIT = 1;
+    LD4_PORT_BIT = 1;
+    
+    //test LEDs by this loop
+    /*
+    int j = 0;
+    for (j=0; j<20; j++) {
+        Delay(8000000);
+        LD1_PORT_BIT = ~ LD1_PORT_BIT;
+        LD2_PORT_BIT = ~ LD2_PORT_BIT;
+        LD3_PORT_BIT = ~ LD3_PORT_BIT;
+        LD4_PORT_BIT = ~ LD4_PORT_BIT;
+    }
+    */
+    
+    // Stop program here
+    //while (1);
+    
+    // BTN1 input
+    // Disable analog mode
+    ANSELAbits.ANSA5 = 0;
```

Comment:
Change this and other similar statements to _bit manipulation_ operation: `ANSEL &= ~BIT5;` or `ANSEL &= ~(1 << 5);`